health/health.go

```diff
@@ -170,6 +170,20 @@ func Register(name string, check Checker) {
 	registeredChecks[name] = check
 }
 
+// Unregister removes the named checker.
+func Unregister(name string) {
+	mutex.Lock()
+	defer mutex.Unlock()
+	delete(registeredChecks, name)
```

Comment:
I'm not expecting anything to call it at runtime. Currently it's only used in tests. But it's a good point that resources would leak with the current implementation.